### ISSUE TITLE
Investigate and document the return values / exceptions for common failure modes

### ISSUE THREAD

**Issue by martinkou** [CoinAlpha/hummingbot#4049]
## Why

This is related to #3991.

Before we can construct mocks for failure cases, we need to know what kind of error responses or exceptions are we getting from the Web3 or Uniswap libraries, when the failure cases happen.

## What

* Document the error response for the following failure modes:
  1. Transaction got stuck due to low gas fee
  2. Transaction failed due to not enough ETH / out of gas.
  3. Computer lost Internet connection
  4. Computer got banned by Infura due to triggering rate limits
  5. (No longer needed) Incorrect contract ABI used for token (e.g. MKR vs. regular ERC20)
  6. (No longer needed) Spender or token are not of the right address
  7. Transaction failed due to any reason other than "out of gas".

**Comment by martinkou** [comment]
After some research yesterday, items 5 and 6 are found to be not needed - there is currently no accessible way for gateway to get to the trace debug messages in a smart contract execution failure, because it requires a full Ethereum node with debug API permissions.

We only need to detect whether a failed transaction failed by out of gas (which can be detected via heuristics), or other reasons. For all other reasons, we should just redirect the user to Tenderly for the actual error from the smart contract.
